### PR TITLE
Homework 3 done

### PR DESCRIPTION
**Pull request by vlad1705**

### CODE REVIEW

File: homework-3-promises-and-worker-threads/task1.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+async function makeFewApiCalls(...links) {
+    let responses  = await Promise.all(links.map((link)=> fetch(link)))
+    // console.log(responses)
+    let jsonedResult = await Promise.all(responses.map(response => response.json()));
```
Comment (Vadym342) [inline homework-3-promises-and-worker-threads/task1.js]: const

File: homework-3-promises-and-worker-threads/task2.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+async function returnFirstResolvedPromise(...promises) {
+    try {
+        let result = await Promise.race(promises);
```
Comment (Vadym342) [inline homework-3-promises-and-worker-threads/task2.js]: We can avoit this variable

File: homework-3-promises-and-worker-threads/task3.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
+
+async function returnAllCompletedPromises(...promises) {
+    const results = await Promise.allSettled(promises);
```
Comment (Vadym342) [inline homework-3-promises-and-worker-threads/task3.js]: The same

File: homework-3-promises-and-worker-threads/task5/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+
+async function readFromFile(filename) {
+   return new Promise((resolve,reject)=>{
```
Comment (Vadym342) [inline homework-3-promises-and-worker-threads/task5/index.js]: I think we can do this without Promise, and we can use in this case readFileSync

File: homework-3-promises-and-worker-threads/task5/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+               data.split("\r\n").forEach((res)=> {
+                   resultMap.set(res.split(":")[0], res.split(":")[1])
+               })
+               resolve(resultMap)
+           }
+       })
+   });
+}
+
+readFromFile("keyValues.txt").then((res)=>console.log(res))
```
Comment (Vadym342) [inline homework-3-promises-and-worker-threads/task5/index.js]: I think it's incorrect response, we need all values in the console). I think response should be like object and we can show it in json format
image

File: homework-3-promises-and-worker-threads/task5/index.js (hunk 4 above)
Comment (vlad1705) [reply homework-3-promises-and-worker-threads/task5/index.js]: Fixed

File: homework-3-promises-and-worker-threads/task3.js (hunk 3 above)
Comment (vlad1705) [reply homework-3-promises-and-worker-threads/task3.js]: Fixed

File: homework-3-promises-and-worker-threads/task2.js (hunk 2 above)
Comment (vlad1705) [reply homework-3-promises-and-worker-threads/task2.js]: Fixed

File: homework-3-promises-and-worker-threads/task1.js (hunk 1 above)
Comment (vlad1705) [reply homework-3-promises-and-worker-threads/task1.js]: Fixed

**Review by Vadym342 — APPROVED** [review]

**Pull request merged**